src/dbus/methods/buddy_whois.c

```diff
@@ -33,32 +33,29 @@ void whois_cb(const struct cmd_whois_data *whois,
               void *args)
 {
     struct cb_args *a = (struct cb_args *) args;
-    const char *gvariant_format = "(sssssuu)";
+
+    GVariantBuilder builder;
     GVariant *result;
 
+    g_variant_builder_init(&builder, G_VARIANT_TYPE("a{sv}"));
+
     if (whois != NULL)
     {
-        result = g_variant_new(gvariant_format,
-                               whois->ip ? whois->ip : "",
-                               whois->country ? whois->country : "",
-                               whois->status, /* todo: int */
-                               whois->profile_id,
-                               whois->online_id,
-                               whois->login_time,
-                               whois->rank);
+        g_variant_builder_add(&builder, "{sv}", "ip", g_variant_new_string(whois->ip ? : ""));
+        g_variant_builder_add(&builder, "{sv}", "country", g_variant_new_string(whois->country ? : ""));
```

Comment:
` ? : ""` is that even compiling? :D
I believe you forgot a `whois->country` there. Same above with IP.

Reply:
yes, it's working good 
http://gcc.gnu.org/onlinedocs/gcc/Conditionals.html#Conditionals

Reply:
This looks like a GNU C extension, not part of C89/C99 standard. If someone tries to recompile wfbot with another compiler, let's say MSVC, it will fail.
https://stackoverflow.com/questions/10946195/in-c-when-is-conditional-test-alt-form-empty-true-case-supported

I prefer using standard C there, sorry.

Reply:
ohh got it okay np

Reply:
done